v2/internal/controller/config/vars.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package config
+
+import (
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	subscriptionIDVar   = "AZURE_SUBSCRIPTION_ID"
+	targetNamespacesVar = "AZURE_TARGET_NAMESPACES"
+	operatorModeVar     = "AZURE_OPERATOR_MODE"
+	podNamespaceVar     = "POD_NAMESPACE"
+)
+
+// Values stores configuration values that are set for the operator.
+type Values struct {
+	// SubscriptionID is the Azure subscription the operator will use
+	// for ARM communication.
+	SubscriptionID string
+
+	// PodNamespace is the namespace the operator pods are running in.
+	PodNamespace string
+
+	// OperatorMode determines whether the operator should run
+	// watchers, webhooks or both.
+	OperatorMode OperatorMode
+
+	// TargetNamespaces lists the namespaces the operator will watch
+	// for Azure resources (if the mode includes running watchers). If
+	// it's empty the operator will watch all namespaces.
+	TargetNamespaces []string
+}
+
+// ReadFromEnvironment loads configuration values from the AZURE_*
+// environment variables.
+func ReadFromEnvironment() (Values, error) {
+	var result Values
+	modeValue := os.Getenv(operatorModeVar)
+	if modeValue == "" {
+		result.OperatorMode = OperatorModeBoth
+	} else {
+		mode, err := ParseOperatorMode(modeValue)
+		if err != nil {
+			return Values{}, err
+		}
+		result.OperatorMode = mode
+	}
+
+	result.SubscriptionID = os.Getenv(subscriptionIDVar)
+	result.PodNamespace = os.Getenv(podNamespaceVar)
+	result.TargetNamespaces = ParseTargetNamespaces(os.Getenv(targetNamespacesVar))
+	// Not calling validate here to support using from tests where we
```

Comment:
Are we actually violating the configs expectations in any of the tests that use `ReadFromEnvironment`? I would've thought that the expectations were only violated in the tests that passed a manually constructed config?

Reply:
The main one is the check for subscription id - since we mostly run the tests in replay mode we don't want to require this to be set in the environment.